packages/firestore/src/remote/backoff.ts

```diff
@@ -92,18 +94,39 @@ export class ExponentialBackoff {
 
     // First schedule using the current base (which may be 0 and should be
     // honored as such).
-    const delayWithJitterMs = this.currentBaseMs + this.jitterDelayMs();
+    const desiredDelayWithJitterMs = Math.floor(
+      this.currentBaseMs + this.jitterDelayMs()
+    );
+
+    // Some time may have already elapsed so account for that.
+
+    // Math.max(0, ...) to guard against lastAttemptTime being in the future due
```

Comment:
nit: rather than repeating the function you're calling, just describe what's happening, i.e.

```js
// Guard against lastAttemptTime being in the future due to a clock change.
```